packages/wallet-sdk/src/sign/scw/SCWSigner.ts

```diff
@@ -22,16 +22,19 @@ type SwitchEthereumChainParam = [
 
 export class SCWSigner implements Signer {
   private readonly metadata: AppMetadata;
+  private readonly waitForPopupLoaded: () => Promise<Window>;
   private readonly postMessageToPopup: (_: RPCRequestMessage) => Promise<RPCResponseMessage>;
   private readonly keyManager: SCWKeyManager;
   private readonly stateManager: SCWStateManager;
 
   constructor(params: {
     metadata: AppMetadata;
+    waitForPopupLoaded: () => Promise<Window>;
```

Comment:
if it needs to access more than one methods of Communicator instance, then let's just pass comm.
or, provider might just open it for scwsigner instead.